com.unity.multiplayer.mlapi/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -46,34 +46,44 @@ internal NetworkSerializer __beginSendServerRpc(uint rpcMethodId, ServerRpcParam
         {
             PooledNetworkWriter writer;
 
-            var rpcQueueContainer = NetworkManager.RpcQueueContainer;
-            var isUsingBatching = rpcQueueContainer.IsUsingBatching();
-            var transportChannel = rpcDelivery == RpcDelivery.Reliable ? NetworkChannel.ReliableRpc : NetworkChannel.UnreliableRpc;
-
-            if (IsHost)
+            if (serverRpcParams.Send.UpdateStage == NetworkUpdateStage.Unset)
             {
-                writer = rpcQueueContainer.BeginAddQueueItemToFrame(RpcQueueContainer.QueueItemType.ServerRpc, Time.realtimeSinceStartup, transportChannel,
-                    NetworkManager.ServerClientId, null, RpcQueueHistoryFrame.QueueFrameType.Inbound, serverRpcParams.Send.UpdateStage);
+                serverRpcParams.Send.UpdateStage = NetworkUpdateLoop.UpdateStage;
 
-                if (!isUsingBatching)
+                if (serverRpcParams.Send.UpdateStage == NetworkUpdateStage.Initialization)
                 {
-                    writer.WriteByte(NetworkConstants.SERVER_RPC); // MessageType
+                    serverRpcParams.Send.UpdateStage = NetworkUpdateStage.EarlyUpdate;
                 }
             }
```

Comment:
This bit of logic seems to be repeating here and in lines 101-111. There are similar repetitions that I will try to mark up - these I would pull out into their own respective functions for better readability and long-term maintainability.

Reply:
Done